### PR TITLE
APP-15504 implement entity multiclass types

### PR DESCRIPTION
**Pull request by ryan-willis**

### CODE REVIEW

File: packages/integration-sdk-core/src/data/createIntegrationHelpers.ts [hunk 1]
```diff
@@ -40,14 +97,14 @@ export const createIntegrationHelpers = <
     ...entityMetadata
   }: Omit<StepEntityMetadata, 'schema'> & {
     resourceName: ResourceName;
-    _class: [Class, ...Class[]];
+    _class: [Class];
```
Comment (jzolo22) [inline packages/integration-sdk-core/src/data/createIntegrationHelpers.ts:100]: 🎉

File: packages/integration-sdk-core/src/data/__tests__/createIntegrationHelpers.test.ts [hunk 2]
```diff
@@ -170,6 +176,255 @@ describe('createIntegrationHelpers', () => {
...
+      {
+        // required on Host
+        message: "must have required property 'hostname'",
+        property: 'hostname',
+        schemaId: '#multi',
+        validation: 'required',
+      },
+      {
+        // required on Multi!!
+        message: "must have required property 'id'",
```
Comment (jzolo22) [inline packages/integration-sdk-core/src/data/__tests__/createIntegrationHelpers.test.ts:357]: 🎉

File: package.json [hunk 3]
```diff
@@ -47,7 +47,7 @@
     "lint-staged": "^10.2.6",
     "prettier": "^3.0.0",
     "ts-jest": "^29.1.1",
-    "typescript": "^5.1.6"
+    "typescript": "^5.5.2"
```
Comment (jzolo22) [inline package.json:50]: What does this version do for us?

**Review by jzolo22 — DISMISSED** [review]

File: package.json (hunk 3 above)
Comment (jzolo22) [reply package.json:50]: Ah nice

File: packages/integration-sdk-core/src/data/__tests__/createIntegrationHelpers.test.ts [hunk 4]
```diff
@@ -170,6 +174,255 @@ describe('createIntegrationHelpers', () => {
     });
   });
 
+  test('createMultiClassEntityMetadata', () => {
+    const [MULTI, createMulti] = createMultiClassEntityMetadata({
+      resourceName: 'Multi',
+      _type: 'multi',
+      _class: [typeboxClassSchemaMap['Device'], typeboxClassSchemaMap['Host']],
```
[the remark this answers is not in the document]
Comment (jzolo22) [reply packages/integration-sdk-core/src/data/__tests__/createIntegrationHelpers.test.ts:181]: 🔥 🔥

**Review by jzolo22 — APPROVED** [review]

**Review by hesto2 — APPROVED** [review]

**Pull request merged**